mesonbuild/interpreter.py

```diff
@@ -2289,9 +2292,16 @@ def func_project(self, node, args, kwargs):
             oi.process(self.option_file)
             self.coredata.merge_user_options(oi.options)
 
-        default_options = mesonlib.stringlistify(kwargs.get('default_options', []))
-        default_options = coredata.create_options_dict(default_options)
-        default_options.update(self.default_project_options)
+        # Do not set default_options on reconfigure otherwise it would override
+        # values previously set from command line. That means that changing
+        # default_options in a project will trigger a reconfigure but won't
+        # have any effect.
```

Comment:
This has always been a bad user experience, do you think we could store whether a default option was overridden by the user or not, and if it was not, to change the value if the default value changed?

Reply:
Of course this is unrelated to this issue, but just a general comment.

Reply:
In the PR that add --wipe command line I store options set by user in a separate file. That could be used for solving this.